### PR TITLE
Quarkiverse adjustments / onboarding

### LINKED ISSUE #397: Quarkiverse adjustments
https://github.com/quarkiverse/quarkiverse/wiki#getting-an-extension-onboarded
https://quarkus.io/guides/writing-extensions#ecosystem
https://hub.quarkiverse.io/checklist/#after-the-repository-is-created

- [x] change maven artifact group/name/version
- [x] change java package names
- [x] update README.md and CONTRIBUTING.md
- [x] Create an Ecosystem CI issue
- [x] Create a quarkiverse-resteasy-problem/info.yaml file in Ecosystem CI repository
- [x] make Ecosystem CI run green
- [x] tag/branch 3.9 for critical bugfixes of the standalone version
- [x] Publish binaries to maven repository
- [x] Add extension to quarkus-extension-catalog repository

### PR DESCRIPTION
**Pull request by lwitkowski**
- changed maven project/artifact to `io.quarkiverse.resteasy-problem:quarkus-resteasy-problem:999-SNAPSHOT`
- changed package names to `io.quarkiverse.resteasy.problem`
- updated readme and contributing
- updated scm definitions
- preparations for master -> main migration
- quarkiverse ecosystem ci integration

### CODE REVIEW

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: Integration Tests
 
 on:
   push:
-    branches: [ master ]
+    branches: [ main ]
```
Comment (pazkooda) [inline .github/workflows/integration-tests.yaml:5]: KInda dislike "politically correct" branch naming... 😞

Comment (lwitkowski) [reply .github/workflows/integration-tests.yaml:5]: This change has nothing to do with p.correctness, and everything with consistency... I looked at few random quarkiverse extensions, and quarkus itself, and all use `main`, so I assume some Ecosystem CI jobs by default target `main` branch.

**Comment by lwitkowski** [comment]
@pazkooda ready for review if you have some time to scan the changes. The real testing will start after merging this to main (in order to test ecosystem ci workflow I need it in the `main`)

File: README.md [hunk 2]
```diff
@@ -48,6 +35,18 @@ so-called "HTTP APIs" are usually not.
 ```
 
 ## Usage
+### Quarkus 3.15+
+Quarkus is now part of Quarkiverse, so simply add this to your pom.xml:
```
Comment (pazkooda) [inline README.md:39]: Quarkus is not a part of Quarkiverse, this extension is ;)

File: deployment/pom.xml [hunk 3]
```diff
@@ -3,17 +3,17 @@
...
+        <version>999-SNAPSHOT</version>
     </parent>
 
     <artifactId>quarkus-resteasy-problem-deployment</artifactId>
     <name>Quarkus - RESTeasy - Problem - Deployment</name>
 
     <dependencies>
         <dependency>
-            <groupId>com.tietoevry.quarkus</groupId>
+            <groupId>io.quarkiverse.resteasy-problem</groupId>
```
Comment (pazkooda) [inline deployment/pom.xml:16]: Just asking: is `io.quarkiverse.resteasy` restricted somehow?
I'd prefer to have `groupId` like ☝️ (avoiding word redudancy), or maybe at least `io.quarkiverse.resteasy.problem` 🤔

File: integration-test/pom.xml [hunk 4]
```diff
@@ -3,23 +3,19 @@
     <modelVersion>4.0.0</modelVersion>
 
     <parent>
-        <groupId>com.tietoevry.quarkus</groupId>
+        <groupId>io.quarkiverse.resteasy-problem</groupId>
         <artifactId>quarkus-resteasy-problem-parent</artifactId>
-        <version>3.9.1-SNAPSHOT</version>
+        <version>999-SNAPSHOT</version>
```
Comment (pazkooda) [inline integration-test/pom.xml:8]: Should this `999` stay here?

**Review by pazkooda — COMMENTED** [review]
Haven't found any major concerns.
Please take a look a minor issue with docs and see if you know answer to my questions 🙂 .

File: README.md (hunk 2 above)
Comment (lwitkowski) [reply README.md:39]: Good catch. The versioning (and docs) will need some polish after we release first version, I'm not sure if quarkiverse-bom which stores versions of the extensions is included by default in quarkus-parent or it needs to be explicitly added by developers - to be tested later.

File: deployment/pom.xml (hunk 3 above)
Comment (lwitkowski) [reply deployment/pom.xml:16]: I had similar thoughts on this, but there seems to be some consensus on this in the docs: https://hub.quarkiverse.io/checklistfornewprojects/#start-coding
```
mvn io.quarkus.platform:quarkus-maven-plugin:create-extension \
 -DgroupId=io.quarkiverse.[myext] -DextensionId=[my-ext] -U
```
If we assume our extension id is `resteasy-problem` then it gives us:
```
<groupId>io.quarkiverse.resteasy-problem</groupId>
<artifactId>quarkus-resteasy-problem</artifactId>
```

this name is used in few other places like ecosystem ci repo, tooling etc

File: integration-test/pom.xml (hunk 4 above)
Comment (lwitkowski) [reply integration-test/pom.xml:8]: That's quarkus and quarkiverse convention everywhere.

File: README.md (hunk 2 above)
Comment (lwitkowski) [reply README.md]: Fixed

**Review by pazkooda — APPROVED** [review]

**Pull request merged**